core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -77,24 +77,27 @@
   public static LocalCacheManager create(AlluxioConfiguration conf) throws IOException {
     MetaStore metaStore = MetaStore.create();
     CacheEvictor evictor = CacheEvictor.create(conf);
-    PageStore pageStore = PageStore.create(conf);
+    PageStoreOptions options = PageStoreOptions.create(conf);
+    PageStore pageStore = PageStore.create(options);
+    boolean restored = false;
     try {
-      Collection<PageInfo> pageInfos = pageStore.getPages();
-      LOG.info("Creating LocalCacheManager with {} existing pages", pageInfos.size());
-      for (PageInfo pageInfo : pageInfos) {
-        PageId pageId = pageInfo.getPageId();
-        metaStore.addPage(pageId, pageInfo);
-        evictor.updateOnPut(pageId);
-      }
-      return new LocalCacheManager(conf, metaStore, pageStore, evictor);
+      restored = pageStore.restore(pageInfo -> {
+        if (pageInfo == null) {
+          return false;
+        }
+        metaStore.addPage(pageInfo.getPageId(), pageInfo);
+        evictor.updateOnPut(pageInfo.getPageId());
+        return metaStore.bytes() <= pageStore.getCacheSize();
+      });
     } catch (Exception e) {
-      try {
-        pageStore.close();
-      } catch (Exception ex) {
-        e.addSuppressed(ex);
-      }
-      throw new IOException("failed to create local cache manager", e);
+      LOG.error("Failed to restore PageStore", e);
+    }
+    if (!restored) {
+      metaStore.reset();
+      evictor.reset();
+      pageStore.initialize(options);
```

Comment:
RocksPageStore will be in bad state after restore then initialize.

Reply:
initialize will remove all files. will it still be a problem?

Reply:
The issue is that initialize is generic and it does not initialize resource specific to an implementation. In RocksDB page store, the mDB will be stale (depends on the restore result) and unusable after the initialize clean up the directory.